NodeJS/src/AgentClient.ts

```diff
@@ -40,7 +51,22 @@ export class AgentClient {
 
         return this.getAgentAndStation(agentUsername);
     }
+    async setDefaultSkillNumberIfNotExists() {
+        let defaultSkillNumber = await this.fetchDefaultSkillNumber()
 
+        if (!!defaultSkillNumber) {
+            this.defaultSkillNumber = defaultSkillNumber
+            return true
+        } else {
+            logger.debug("tring to create default skill")
+            let created = await this.createDefaultSkill()
+            if(!created){
+                logger.debug("failed to submit job to create default skill")
+                return false
+            }
+            return await this.waitForDefaultSkillCreation()  
+        }
+    }
```

Comment:
I would re-write this method, it's small but confusing. It has nested `if`s, multiple exit points, and the the method name is not properly descriptive of what it does.

The method name is `setDefaultSkillNumberIfNotExists`, which implies to me that this should only be called after asserting that the Skill Number does not exist. But the first thing the function does is fetch the default skill, set it, and return true. Else it tries to create it and returns false if it can't, else it polls until it sees the new skill and returns a bool based on that; at least that's what I think it's doing.

If I understood this method correctly, I'd re-write it like:

```
async setDefaultSkillNumber(): boolean {
  this.defaultSkillNumber = this.defaultSkillNumber || await this.fetchDefaultSkillNumber()
  if (!!this.defaultSkillNumber) return true

  return await this.createDefaultSkill() ? await this.waitForDefaultSkillCreation() : false
}
```

Reply:
Now that I've written all of that, this is really just a "nit". It's a tiny function and it only took me a moment to figure it out. But I do think that my code is easier to read (as I suppose everyone always does).

Reply:
The return line is compact because you removed logging statement.  The defaultSkillNumber line is however might not be right. Since this.defaultSkillNumber starts with -1.